pandas/core/internals/blocks.py

```diff
@@ -2228,6 +2251,9 @@ def set(self, locs, values):
     def external_values(self):
         return np.asarray(self.values.astype("datetime64[ns]", copy=False))
 
+    def array_values(self) -> ExtensionArray:
```

Comment:
these can be generic no? as the EA class name is known by the block

Reply:
Only for everything that subclasses ExtensionBlock, I assume? 
Yes, that would be possible, but 1) not all EAs have a `_simple_new` (which is essential here to have good performance) and 2) other blocks already have their `.values` as EA, and I would like to avoid the overhead going through the EA constructor unneeded.